platform-sdk/swirlds-common/src/main/java/com/swirlds/common/metrics/platform/DefaultLongAccumulator.java

```diff
@@ -22,37 +22,42 @@
 import com.swirlds.common.metrics.LongAccumulator;
 import com.swirlds.common.metrics.platform.Snapshot.SnapshotEntry;
 import java.util.List;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.LongBinaryOperator;
+import java.util.function.LongSupplier;
 import org.apache.commons.lang3.builder.ToStringBuilder;
 
 /**
  * Platform-implementation of {@link LongAccumulator}
  */
 public class DefaultLongAccumulator extends DefaultMetric implements LongAccumulator {
 
-    private final java.util.concurrent.atomic.LongAccumulator container;
-    private final long initialValue;
+    private final AtomicLong container;
+    private final LongBinaryOperator accumulator;
+    private final LongSupplier initializer;
 
     public DefaultLongAccumulator(final LongAccumulator.Config config) {
         super(config);
-        this.container =
-                new java.util.concurrent.atomic.LongAccumulator(config.getAccumulator(), config.getInitialValue());
-        this.initialValue = config.getInitialValue();
+        this.accumulator = config.getAccumulator();
+        final long initialValue = config.getInitialValue();
+        this.initializer = config.getInitializer() != null ? config.getInitializer() : () -> initialValue;
```

Comment:
Since config is not a record 2 calls to config.getInitializer() could result in 2 different values. in worst case the second call could return null. Therefore I would refactor this to only have 1 call to config.getInitializer()